hack/generate/samples/internal/go/v2/memcached_with_webhooks.go

```diff
@@ -90,7 +90,8 @@ func (mh *MemcachedGoWithWebhooks) Run() {
 	pkg.CheckError("scaffolding webhook", err)
 
 	mh.implementingWebhooks()
-	mh.uncommentKustomizationFile()
+	mh.uncommentDefaultKustomization()
+	mh.uncommentManifestsKustomization()
```

Comment:
Our goal is to integrate the projects with OLM. So, I think is very fine we try to solve that by default. 
However, I'd like to raise some concerns about how we are doing that. 

- Why we do not change the sdk command which generates the OLM file to remove the amount from the bundle instead of changing the default scaffold? Is not it only a problem for OLM since it is not supported? 
- Why are we adding things that need to be removed manually after when the CERT starts to be supported? Why we are increasing the complexity of the kustomize file instead of only solve the issue for / where / when it is a problem until we are able to address this need? then, see that when we start to support cert-manager we will not need for the users to change their projects manually again. 
- WDYT about when the bundle be generated and the amount be removed it be informed to the user by a warning output with the link for the issue? IMO we need to clarify that it is not supported and provide a way for the follow up the issue as well. 

/hold

WDYT @estroz @varshaprasad96 @jmrodri

Reply:
* The mount is only a problem for OLM, since it does not support cert-manager yet. Changing `config/default` would alter kubebuilder behavior, which we don't want. The whole point of kustomize is to create a hierarchy of YAML files and combine them in different ways for different environments, which `config/manifests` does with minimal changes.
* This is a problem _now_. Webhooks do not work when deployed via OLM. See https://github.com/operator-framework/operator-sdk/issues/4244#issuecomment-732355073.
* The UX you suggest is way worse than just patching `kustomize build config/default` output. Plus that would break "vanilla" operator deployment. OLM already [documents](https://olm.operatorframework.io/docs/advanced-tasks/adding-admission-and-conversion-webhooks/#certificate-authority-requirements) that it doesn't support cert-manager. This will be linked in the docs I intend to write before this PR is merged.

>I think we need to be very careful by addressing workarounds that require manual changes about changes that affect the files and users generated by the tool and the users. Have we not another way to solve that?

There is no other way to solve this problem short of OLM supporting cert-manager _with minimal changes_. You could replicate `config/default` in `config/manifests` (indeed, I [tried that](https://github.com/estroz/operator-sdk/commit/50c90f85ecd727a738b3c64ffeb535c99a5ff721)), but this path creates more places for users to update YAML despite being more kustomize-y.

Reply:
Can we not make the opertor-sdk generate bundle remove it when gen the csv?

Reply:
Then `generate bundle` is (poorly) emulating kustomize behavior, more so than the command already is. How will it know which mount to remove if a user changes its name? What if more cert-manager “stuff” is added later to kustomize configs that also needs to be removed?

Reply:
Fair point 👍 
/hold cancel